astropy/utils/masked/function_helpers.py

```diff
@@ -571,6 +571,11 @@ def bincount(x, weights=None, minlength=0):
 
     @dispatched_function
     def msort(a):
+        if not NUMPY_LT_1_24:
+            warnings.warn(
+                "msort is deprecated, use np.sort(a, axis=0) instead",
+                DeprecationWarning,
```

Comment:
This is missing a `stacklevel`, but the call goes through `<__array_function__ internals>`, which apparently destroys Python's ability to do anything sensible with the `stacklevel`. This warning is only displayed for users that have `astropy` > 7.1.0 together with 1.24 <= `numpy` < 2.0. I would imagine most users have either a recent version of both or an old version of both, so the number of people that have the potential of seeing this warning is small enough that the `stacklevel` is not worth worrying about too much.